streaming/base/batching/per_stream.py

```diff
@@ -94,7 +94,13 @@ def generate_work_per_stream_batching(dataset: StreamingDataset, world: World, e
             logger.warning(
                 'Because of the `per_stream` batching method, some batches with an inadequate \
                 number of samples from stream with index ' + str(stream_idx) + ' will be dropped.')
-        batches_from_partitions.append(global_batches_inorder[:num_full_batches])
+        if num_full_batches > 0:
+            batches_from_partitions.append(global_batches_inorder[:num_full_batches])
+        else:
+            logger.warning('Stream with index ' + str(stream_idx) +
```

Comment:
Use f-string and `+` to concatenate long strings like [this](https://github.com/mosaicml/streaming/blob/main/streaming/base/dataset.py#L388)?